engine/engine_impl.go

```diff
@@ -28,15 +29,18 @@ import (
 )
 
 var backoff = wait.Backoff{
-	Steps:    5,
+	Steps:    15,
 	Duration: 500 * time.Millisecond,
 	Factor:   1.0,
-	Jitter:   0.1,
+	Jitter:   0.5,
 }
 
 // Kubernetes implements a Kubernetes pipeline engine.
 type Kubernetes struct {
 	client *kubernetes.Clientset
+	// Protects concurrent calls to PodInterface.Update to reduce the
+	// chance of a self-inflicted concurrent modification error.
+	updateMutex *sync.Mutex
```

Comment:
mutex does not need to be a pointer

Reply:
also do we need a global mutex (all pods) or just a single mutex per pod?

Reply:
Thanks! Didn't do enough digging to realize we never use `Kubernetes` by value.

To answer second question: we only need a mutex per pod. Should I put it in `engine.Spec`, or is that totally the wrong place for it?

Reply:
good idea, yes, I think that could be nice